### PR TITLE
Create text2vec-cohere module

### PR DESCRIPTION
**Pull request by DasithEdirisinghe**
### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: modules/text2vec-cohere/neartext/searcher.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+	}
+
+	if len(objects) > 0 {
+		var id strfmt.UUID
+		for _, obj := range objects {
+			if len(obj.ID) > 0 {
+				id = strfmt.UUID(obj.ID)
+			}
+			if len(obj.Beacon) > 0 {
+				ref, err := crossref.Parse(obj.Beacon)
```
Comment (antas-marcin) [inline modules/text2vec-cohere/neartext/searcher.go:123]: parsed `ref` can have a `Class` information in it, so if `ref.Class != ""` please pass this information to `findVectorFn` method like here the `id` is passed

Comment (DasithEdirisinghe) [reply modules/text2vec-cohere/neartext/searcher.go:122]: Resolved

File: modules/text2vec-cohere/vectorizer/class_settings.go [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+	if ic.cfg == nil {
+		// we would receive a nil-config on cross-class requests, such as Explore{}
+		return defaultValue
+	}
+
+	model, ok := ic.cfg.Class()[name]
+	if ok {
+		asString, ok := model.(string)
+		if ok {
+			return strings.ToLower(asString)
```
Comment (antas-marcin) [inline modules/text2vec-cohere/vectorizer/class_settings.go:150]: you should remove `strings.ToLower` method here and just return a string, it breaks setting of `truncate` value as the value being returned is always lowercased and you are checking against `availableTruncates` array which holds upper cased string values

File: modules/text2vec-cohere/vectorizer/class_settings.go [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+const (
+	DefaultCohereModel           = "large"
+	DefaultTruncate              = "NONE"
+	DefaultVectorizeClassName    = true
+	DefaultPropertyIndexed       = true
+	DefaultVectorizePropertyName = false
+)
+
+var availableCohereModels = []string{"small", "medium", "large"}
+var availableTruncates = []string{"NONE", "LEFT", "RIGHT"}
```
Comment (antas-marcin) [inline modules/text2vec-cohere/vectorizer/class_settings.go:34]: In order to get rid of the linter error, you can refactor this part to:
```
var (
	availableCohereModels = []string{"small", "medium", "large"}
	availableTruncates    = []string{"NONE", "LEFT", "RIGHT"}
)
```

**Comment by DasithEdirisinghe** [comment]
I agree to CLA

**Review by antas-marcin — APPROVED** [review]

**Pull request merged**